### PR TITLE
verbs: Add support for fi_trywait call.

### PR DESCRIPTION
**Pull request by a-ilango**

### CODE REVIEW

**Comment by shefty** [comment]
Can you please re-push this PR, so that it can get a fresh travis run?

**Comment by a-ilango** [comment]
Done.

File: prov/verbs/src/verbs_domain.c [hunk 1]
```diff
@@ -256,6 +256,36 @@ fi_ibv_domain(struct fid_fabric *fabric, struct fi_info *info,
...
+		case FI_CLASS_CNTR:
+		case FI_CLASS_WAIT:
+			return -FI_ENOSYS;
+		default:
+			return -FI_EINVAL;
+		}
+
+		ret = trywait(fids[i]);
+		if (ret)
+			return ret;
```
Comment (shefty) [inline prov/verbs/src/verbs_domain.c]: I would drop the trywait variable and just call cq->trywait directly in the switch statement.

Comment (a-ilango) [inline prov/verbs/src/verbs_domain.c]: Done.

File: prov/verbs/src/verbs_cq.c [hunk 2]
```diff
@@ -345,6 +345,53 @@ int fi_ibv_cq_signal(struct fid_cq *cq)
...
+
+	rc = fi_ibv_poll_cq(cq, &wce->wc);
+	if (rc > 0) {
+		slist_insert_tail(&wce->entry, &cq->wcq);
+		goto out;
+	} else if (rc < 0) {
+		goto err;
+	}
+
+	rc = ibv_req_notify_cq(cq->cq, 0);
```
Comment (shefty) [inline prov/verbs/src/verbs_cq.c]: Where are the CQ events read from the kernel and acked?

**Comment by a-ilango** [comment]
Added call to read CQ events in trywait and to improve performance acknowledgement of CQ events is delayed until the CQ is closed. Refactored cq_sread to make use of fi_trywait

**Comment by shefty** [comment]
:+1:

**Pull request merged**